src/main/java/seedu/address/model/person/Tag.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+enum Tags {
+    NR("No Reply"),
+    OA("Online Assessment"),
+    I("Interview"),
+    R("Reject"),
+    O("Offer");
+
+    public final String descriptiveName;
+
+    Tags(String descriptiveName) {
+        this.descriptiveName = descriptiveName;
+    }
+
+    // Method to check if a given string is a valid enum value
+    public static boolean isValidTag(String input) {
+        for (Tags tag : Tags.values()) {
+            if (tag.name().equalsIgnoreCase(input)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // Method to find a tag by string
+    public static Tags fromString(String input) {
+        for (Tags tag : Tags.values()) {
+            if (tag.name().equalsIgnoreCase(input) || tag.descriptiveName.equalsIgnoreCase(input)) {
+                return tag;
+            }
+        }
+        throw new IllegalArgumentException("Invalid tag: " + input);
+    }
+}
```

Comment:
Good use of enums here ! I like the use of methods to check the validity as well as retrieving the tags ! LGTM 👍🏾